source/common/io/io_uring_worker_impl.h

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include "envoy/common/io/io_uring.h"
+#include "envoy/event/file_event.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/common/linked_object.h"
+#include "source/common/common/logger.h"
+#include "source/common/io/io_uring_impl.h"
+
+namespace Envoy {
+namespace Io {
+
+class IoUringWorkerImpl;
+
+class IoUringSocketEntry : public IoUringSocket,
+                           public LinkedObject<IoUringSocketEntry>,
+                           public Event::DeferredDeletable,
+                           protected Logger::Loggable<Logger::Id::io> {
+public:
+  IoUringSocketEntry(os_fd_t fd, IoUringWorkerImpl& parent);
+
+  // IoUringSocket
+  os_fd_t fd() const override { return fd_; }
+  void injectCompletion(uint32_t type) override;
+
+  // This will cleanup all the injected completions for this socket and
+  // unlink itself from the worker.
+  void cleanup();
+  void onAccept(int32_t, bool injected) override {
+    if (injected && (injected_completions_ & RequestType::Accept)) {
+      injected_completions_ &= ~RequestType::Accept;
+    }
+  }
+  void onClose(int32_t, bool injected) override {
+    if (injected && (injected_completions_ & RequestType::Close)) {
+      injected_completions_ &= ~RequestType::Close;
+    }
+  }
+  void onCancel(int32_t, bool injected) override {
+    if (injected && (injected_completions_ & RequestType::Cancel)) {
+      injected_completions_ &= ~RequestType::Cancel;
+    }
+  }
+  void onConnect(int32_t, bool injected) override {
+    if (injected && (injected_completions_ & RequestType::Connect)) {
+      injected_completions_ &= ~RequestType::Connect;
+    }
+  }
+  void onRead(int32_t, bool injected) override {
+    if (injected && (injected_completions_ & RequestType::Read)) {
+      injected_completions_ &= ~RequestType::Read;
+    }
+  }
+  void onWrite(int32_t, bool injected) override {
+    if (injected && (injected_completions_ & RequestType::Write)) {
+      injected_completions_ &= ~RequestType::Write;
+    }
+  }
+
+private:
+  void unlink();
+
+  os_fd_t fd_;
+  IoUringWorkerImpl& parent_;
+  uint32_t injected_completions_{0};
+};
+
+class IoUringWorkerImpl : public IoUringWorker, protected Logger::Loggable<Logger::Id::io> {
```

Comment:
If no subclass that needs logging I think it can inherit private from logger (here and above)

Reply:
got it